src/main/host/host.rs

```diff
@@ -490,27 +483,37 @@ impl Host {
         self.params.requested_bw_down_bits / (8 * 1024)
     }
 
-    /// Run `f` with a reference to the network interface associated with
-    /// `addr`, or returns `None` if there is no such interface.
+    /// Returns `None` if there is no such interface.
+    ///
+    /// Panics if we have not yet initialized our network interfaces yet.
+    pub fn interface_mut(
+        &self,
+        addr: Ipv4Addr,
+    ) -> Option<impl Deref<Target = NetworkInterface> + DerefMut + '_> {
+        match self.interface_cell(addr) {
+            Some(rc) => {
+                let interface = rc.borrow_mut();
+                Some(RefMut::map(interface, |i| i.as_mut().unwrap()))
+            }
+            None => None,
+        }
+    }
```

Comment:
Just an idea, but maybe worth adding a `#[track_caller]` to these? If anything in these functions panic, it would probably be more useful to know where the function was called from, rather than that for example the specific `borrow_mut()` function panicked.

Reply:
makes sense; done